src/librustc_mir/borrow_check/nll/constraint_generation.rs

```diff
@@ -164,7 +165,9 @@ impl<'cg, 'cx, 'gcx, 'tcx> Visitor<'tcx> for ConstraintGeneration<'cg, 'cx, 'gcx
                 self.location_table.mid_index(location),
             ));
 
-            for successor_block in terminator.successors() {
+            let mut successor_blocks = terminator.successors();
+            all_facts.cfg_edge.reserve(successor_blocks.by_ref().count());
+            for successor_block in successor_blocks {
```

Comment:
> Yes, but how? I only changed size_hint to count (plus the related iterator stuff).

`.count()` consumes the iterator and you're trying to iterate over it after consuming it here.

Reply:
Ow, I forgot that `by_ref` still advances the iterator; I'll revert that.